src/sgame/sg_main.cpp

```diff
@@ -2858,6 +2858,11 @@ void G_RunFrame( int levelTime )
 		}
 	}
 
+	// ThinkingComponent should have been called already but who knows maybe we forgot some.
```

Comment:
Instead of this, do you think we should include a  frame counter and make sure the think counter matches the frame counter? Throw an assert instead or a log telling people that they should report this. I did this one time, and I got a bunch of bugreport. Otherwise, should prolly be ok since we shouldn't be thinking if enough time hasn't passed right?

Reply:
I think the intention with the ForEntities above was to call all the new-style thinkers before G_RunThink, so around line 2814.

Reply:
That would make sense. I was afraid of breaking something by changing the order of execution, but I guess I'll go ahead and try it.

Reply:
Never mind, actually I don't think it makes sense. Look at the stuff about "free before thinking" and "free after thinking" around line 2600.